specs/WindowControlsOverlayConfiguration.md

```diff
@@ -0,0 +1,142 @@
+WebView2 Window Controls
+===
+
+# Background
+This API allows you to Enable and configure The Webview2 Window Controls overlay.
+The Overlay is a region on the top right/left of the webview window which contains 
+the caption buttons (minimize, maximize, restore, close). Enabing the Overlay allows
+for custom app title bars rendered completly inside the webview window.
+The overlay Settings lives on the controller object.
+
+This API is designed to be used in addition with the other non-client region APIs 
+and features. These include `app-region: drag`, and `IsNonClientRegionSupportEnabled`.
+# Examples
+
+## Win32 C++
+```cpp 
+AppWindow::AppWindow() {
+    m_mainWindow = CreateWindowExW(
+        WS_EX_CONTROLPARENT, 
+        GetWindowClass(),
+        L""
+        WS_POPUPWINDOW, 
+        0,0, 800, 800, 
+        nullptr, nullptr, 
+        g_hInstance, nullptr);
+}
+
+void  AppWindow::OnCreateWebview2ControllerCompleted(HRESULT hr, ICoreWebview2Controller* controller) 
+{
+    wil::com_ptr<ICoreWebView2Controller5> controller5;
+    CHECK_FAILURE(controller->QueryInterface(&controller5));
+
+    wil::com_ptr<ICoreWebView2WindowControlsOverlaySettings> wco_config;
+    CHECK_FAILURE(controller5->get_WindowControlsOverlaySettings(&wco_config));
+
+    wco_config->put_IsEnabled(true);
+    COREWEBVIEW2_COLOR color {1, 0, 0, 225};
+    wco_config->put_TitleBarColor(color);
+}
+```
+## .NET C#
+```c#
+// WebView2 control is defined in the xaml
+// <wv2:WebView2 x:Name="webView" Source="https://www.microsoft.com/"/>
+public MainWindow() 
+{
+    InitializeComponent();
+    m_AppWindow.TitleBar.ExtendsContentIntoTitleBar = true;
+
+    CoreWebView2WindowControlsOverlaySettings config = _coreWebView2Controller.WindowControlsOverlaySettings;
```

Comment:
We don't have a way to get from a WebView2 class to the controller. Where does _coreWebView2Controller come from? I would expect we either plan to update WPF / WinForms / WinUI2/3 WebView2 class to add something for WindowControlsOverlay or to say your new API is only supported when directly using the CoreWebView2Controller class (eg not using a WebView2 class)

Reply:
@tochukwuIbeEkeocha, can you remind we why we decided to put this object on the Controller and not CoreWebView2? I recall at one point we were thinking about passing in the setting in the ControllerOptions, was that why?

Controller is meant for things the framework needs to host a WebView2, or for things where the framework element already has a defined property (i.e. BackgroundColor). The WindowControlOverlay is content being shown inside the WebView2. If we put it on the CoreWebView2, then it is just available in all the frameworks without us and WinUI having to do work to each WebView2 element.

Reply:
Will move it to the the CoreWebview2 Object.